extensions/scarb-snforge-test-collector/src/compilation/test_collector/config.rs

```diff
@@ -284,60 +287,56 @@ fn extract_fork_config_from_args(db: &dyn SyntaxGroup, attr: &Attribute) -> Opti
         return None;
     }
 
-    let block_id_type = block_id
-        .path(db)
-        .elements(db)
-        .last()
-        .unwrap()
-        .identifier(db)
-        .to_string();
+    let block_id_type = elements[1].clone();
 
     let args = block_id.arguments(db).arguments(db).elements(db);
     let expr = match args.first()?.arg_clause(db) {
         ArgClause::Unnamed(unnamed_arg_clause) => Some(unnamed_arg_clause.value(db)),
         _ => None,
     }?;
-    let block_id = try_get_block_id(db, &block_id_type, &expr)?;
+    let block_id_value = try_get_block_id(db, &block_id_type, &expr)?;
 
-    Some(RawForkConfig::Params(RawForkParams { url, block_id }))
+    Some(ForkConfigString::Params(ForkParamsString {
+        url,
+        block_id_type,
+        block_id_value,
+    }))
 }
 
-fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<BlockId> {
+fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<String> {
     match block_id_type {
         "Number" => {
             if let Expr::Literal(value) = expr {
-                return Some(BlockId::Number(
-                    u64::try_from(value.numeric_value(db).unwrap()).ok()?,
-                ));
+                return Some(
+                    u64::try_from(value.numeric_value(db).unwrap())
+                        .ok()?
+                        .to_string(),
+                );
             }
-            None
         }
         "Hash" => {
+            // TODO #1179: add range check
```

Comment:
https://github.com/foundry-rs/starknet-foundry/issues/1179

Reply:
Why do we need to have it here? Wouldn't be enough if foundry did validations on your side?

Reply:
Not really since we wouldn't be able to show the exact place where the error occured using Cairo diagnostics